src/abstracts.ts

```diff
@@ -148,7 +158,7 @@ export interface BlockQueryProperties {
 export interface VariableProperties {
     id: string;
     name: string;
-    type?: string;
-    isCloud: boolean;
+    type: string;
+    isCloud?: boolean; // this only applies to scalar type varibles
```

Comment:
I >really< prefer comments before the line they comment not at the end of it - mind doing this little nit?

Reply:
There should be a lint we can turn on?